src/probability/martingale/convergence.lean

```diff
@@ -231,4 +251,242 @@ mem_ℒp_limit_process_of_snorm_bdd
+      with x heq ht,
+    rwa heq },
+  have : ∀ n, ∀ s, measurable_set[ℱ n] s → ∫ x in s, g x ∂μ =
+    ∫ x in s, ℱ.limit_process (λ n x, μ[g | ℱ n] x) μ x ∂μ,
+  { intros n s hs,
+    rw [← set_integral_condexp (ℱ.le n) hintg hs, ← set_integral_condexp (ℱ.le n) hlimint hs],
+    refine set_integral_congr_ae (ℱ.le _ _ hs) _,
+    filter_upwards [(martingale_condexp g ℱ μ).ae_eq_condexp_limit_process hunif n] with x hx _,
+    rwa hx },
+  refine ae_eq_of_forall_set_integral_eq_of_sigma_finite' hle
+    (λ s _ _, hintg.integrable_on) (λ s _ _, hlimint.integrable_on) (λ s hs, _)
+    hgmeas.ae_strongly_measurable' strongly_measurable_limit_process.ae_strongly_measurable',
+  have hgen : (⨆ n, ℱ n) = measurable_space.generate_from {s | ∃ n, measurable_set[ℱ n] s},
+  { ext s,
+    rw measurable_space.measurable_set_Sup,
+    simp_rw set.mem_range,
+    change _ ↔ measurable_space.generate_measurable {s | ∃ n, measurable_set[ℱ n] s} s,
+    simp only [exists_prop, exists_exists_eq_and] },
+  refine @measurable_space.induction_on_inter _ _ _ (⨆ n, ℱ n) hgen _ _ _ _ _ _ hs,
+  { rintro s ⟨n, hs⟩ t ⟨m, ht⟩ -,
+    by_cases hnm : n ≤ m,
+    { exact ⟨m, (ℱ.mono hnm _ hs).inter ht⟩ },
+    { exact ⟨n, hs.inter (ℱ.mono (not_le.1 hnm).le _ ht)⟩ } },
+  { simp only [measure_empty, with_top.zero_lt_top, measure.restrict_empty,
+      integral_zero_measure, forall_true_left] },
+  { rintro t ⟨n, ht⟩ -,
+    exact this n _ ht },
+  { rintro t htmeas ht -,
+    have hgeq := @integral_add_compl _ _ (⨆ n, ℱ n) _ _ _ _ _ _ htmeas (hintg.trim hle hgmeas),
+    have hheq := @integral_add_compl _ _ (⨆ n, ℱ n) _ _ _ _ _ _ htmeas
+      (hlimint.trim hle strongly_measurable_limit_process),
+    rw [add_comm, ← eq_sub_iff_add_eq] at hgeq hheq,
+    rw [set_integral_trim hle hgmeas htmeas.compl,
+      set_integral_trim hle strongly_measurable_limit_process htmeas.compl,
+      hgeq, hheq, ← set_integral_trim hle hgmeas htmeas,
+      ← set_integral_trim hle strongly_measurable_limit_process htmeas,
+      ← integral_trim hle hgmeas, ← integral_trim hle strongly_measurable_limit_process,
+      ← integral_univ, this 0 _ measurable_set.univ, integral_univ, ht (measure_lt_top _ _)] },
+  { rintro f hf hfmeas heq -,
+    rw [integral_Union (λ n, hle _ (hfmeas n)) hf hintg.integrable_on,
+      integral_Union (λ n, hle _ (hfmeas n)) hf hlimint.integrable_on],
+    exact tsum_congr (λ n, heq _ (measure_lt_top _ _)) }
+end
+
+/-- Part c of the **L¹ martingale convergnce theorem**: Given a integrable function `g` which
+is measurable with respect to `⨆ n, ℱ n` where `ℱ` is a filtration, the martingale defined by
+`𝔼[g | ℱ n]` converges in L¹ to `g`.
+
+This martingale also converges to `g` almost everywhere and this result is provided by
+`measure_theory.mem_ℒp.tendsto_ae_condexp` -/
+lemma mem_ℒp.tendsto_snorm_condexp
+  (hg : mem_ℒp g 1 μ) (hgmeas : strongly_measurable[⨆ n, ℱ n] g) :
```

Comment:
please use `integrable g μ` instead of `mem_ℒp g 1 μ`

Reply:
I remember that you were working to replace `integrable` with `mem_ℒp`. Is this no longer the case?